### PR TITLE
add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by obuckley-kenzan**
helps to resolve issue #4 .  All projects will link back to this document.

cc: @thisgeek / @paul-barry-kenzan / @thisgeek / @csharon / @jpoo / @rkunboxed / @mornindew / @cxoltero / @jon077

### CODE REVIEW

**Comment by csharon** [comment]
Looks good, nice work.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+all contributors will adhere to the [contributor covenant][]
+
+Please make sure you have read the `README` for your given project.
+
+[contributor covenant]: http://contributor-covenant.org/version/1/2/0/
+
+## Pull Requests
+When making a PR, the following criteria should be met:
+
+1. Your PR has an issue created that has been reviewed by the Keystone maintaining team
```
Comment (rkunboxed) [inline CONTRIBUTING.md]: Nitpick: add periods for 1-7.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+1. If you have added any dependencies to your project, links to their repo and an explanation of their need should 
+be provided
+2. Please make sure your dependencies are installed correctly.  `dependencies` are those modules that are explicitly
+needed by this package in order to run.  `devDependencies` are only needed for local development or building the 
+package.  The rule of thumb is, _if it goes in src/index.js it's a dependency, otherwise it's a devDependency_
+3. NPM package versions should not contain `~` or `^` as we only want to pin to specific versions.  They should be
+organized alpha sorted ascending.
+
+## Commit Messages
+Commit message are important when looking at the work history of an application.  Please consider the following as you
```
Comment (rkunboxed) [inline CONTRIBUTING.md]: "messages" plural

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+## Commit Messages
+Commit message are important when looking at the work history of an application.  Please consider the following as you 
+commit your work:
+
+1. Use the present tense ("Add feature" not "Added feature")
+2. Use the imperative mood ("Move cursor to..." not "Moves cursor to...")
+3. Limit the first line to 72 characters or less
+4. Reference issues and pull requests liberally
+
+All PRs will be squashed merged through the GitHub.
```
Comment (rkunboxed) [inline CONTRIBUTING.md]: I think this might read a little better: "All PRs will be squash merged through GitHub."

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+## Continuous Integration
+Keystone projects enforce a number of best practices such as JS Linting and unit testing.  Each PR will be run against
```
Comment (rkunboxed) [inline CONTRIBUTING.md]: I don't think "Linting" needs to be capitalized.

File: CONTRIBUTING.md (hunk 3 above)
Comment (obuckley-kenzan) [inline CONTRIBUTING.md]: @rkunboxed 
Could you clarify or expand on the ambiguity you would like cleared up?

Comment (paul-barry-kenzan) [inline CONTRIBUTING.md]: "*the* GitHub" vs. "GitHub"

Comment (obuckley-kenzan) [inline CONTRIBUTING.md]: oh, duh.  got it.

**Comment by obuckley-kenzan** [comment]
thanks for all your feedback everyone!  We can update this as needed as the project grows

**Pull request merged**